src/main/java/bio/terra/service/resourcemanagement/azure/AzureStorageAccountResource.java

```diff
@@ -100,6 +100,17 @@ public AzureStorageAccountResource region(AzureRegion region) {
     return this;
   }
 
+  public String determineContainer(ContainerType containerType) {
+    switch (containerType) {
+      case DATA:
+        return getDataContainer();
+      case METADATA:
+        return getMetadataContainer();
+      default:
+        throw new IllegalArgumentException("Invalid container type");
+    }
+  }
+
```

Comment:
I really like to avoid switching over enums whenever possible. Here's one approach to fix this case.
```
  public enum ContainerType {
    DATA() {
      String getContainer(AzureStorageAccountResource account) {
          return account.getDataContainer();
      }
    },
    METADATA() {
      String getContainer(AzureStorageAccountResource account) {
          return account.getMetadataContainer();
      }
    };
    abstract String getContainer(AzureStorageAccountResource account);
  }
```

Reply:
I'm curious: why avoid the enum switch?

Reply:
- you don't have to worry about `default` - never need to `throw`, and if a new enum is added, you know right away you have to add a method for it (maintenance)
- the code for different enums is close to the enum, instead of in the code that uses the enum (encapsulation)

Reply:
Of course, encapsulation can work against you too; you don't necessarily want every enum to know about every bit of logic that uses the enum. In this case the enum was created specifically for this class, so it seems like a good candidate.

Reply:
Hmm...fair enough. Ok, lemme change this.  I dig it